src/single-spa-alpinejs.js

```diff
@@ -32,6 +53,49 @@ function bootstrap(opts, props) {
   return Promise.resolve();
 }
 
+/**
+ * Create the Alpine Element
+ * @param {*} template - The final template
+ * @param {*} opts     - alpine options
+ * @param {*} props    - props
+ */
+function createAlpineElement(template, opts, props) {
+  const typeofXData = typeof opts.xData;
+  // create child Element
+  let childElement = document.createElement("div");
+  childElement.id = `alpine-${props.name}`;
+  childElement.innerHTML = template;
+  const appName = props.appName || props.name;
+  // create any global x-init,x-data functions that are needed
+  // setup opts x-data value
+  if (opts.xData) {
+    const originalData =
```

Comment:
To support the `xData()` function returning promises, we can return the value from our `.then()` function and then add another `.then()` for when it finishes resolving. This works when returning both plain values and with promise values. The code below shows the idea

```js
export function mount(opts, props) {
  return Promise.resolve().then(() => {
    if (opts.xData) {
      return typeof opts.xData === 'function' ? opts.xData({...props}) : opts.xData
    } else {
      return {}
    }
  }).then(originalData => {
    const finalData = Object.assign({}, props, originalData)
  })
}
 
```